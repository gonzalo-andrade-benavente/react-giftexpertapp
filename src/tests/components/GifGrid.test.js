import { shallow } from 'enzyme';
import GifGrid from '../../components/GiftGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test GifGrid.test.js', () => {

    const category = 'Naruto';

    test('Should be show the Component Correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should be show Items with the function useFetchGifs', () => {

        const gifs = [
            {
                id: 123456,
                title: 'img.title',
                img: 'https://img.images?.downsized_medium.url'

            },  {
                id: 1234567,
                title: 'img.title',
                img: 'https://img.images?.downsized_medium.url'

            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        
        //expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GiftGridItem').length ).toBe( gifs.length );

    });



});