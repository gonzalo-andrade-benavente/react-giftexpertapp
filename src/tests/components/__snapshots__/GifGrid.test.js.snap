// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid.test.js Should be show the Component Correctly 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <p
    className=" animate__animated animate__flash "
  >
    Loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
